File: src_original/extras-attica/libbuildservicejoboutput.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/attica-buildservicejoboutput.html
pub const attica__buildservicejoboutput = struct {
    /// New constructs a new Attica::BuildServiceJobOutput object.
    ///
    ///
    pub fn New() QtC.Attica__BuildServiceJobOutput {
        return qtc.Attica__BuildServiceJobOutput_new();
    }

    /// New2 constructs a new Attica::BuildServiceJobOutput object.
    ///
    /// ``` other: QtC.Attica__BuildServiceJobOutput ```
    pub fn New2(other: ?*anyopaque) QtC.Attica__BuildServiceJobOutput {
        return qtc.Attica__BuildServiceJobOutput_new2(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#operator-eq)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput, other: QtC.Attica__BuildServiceJobOutput ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.Attica__BuildServiceJobOutput_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#setOutput)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput, output: []const u8 ```
    pub fn SetOutput(self: ?*anyopaque, output: []const u8) void {
        const output_str = qtc.libqt_string{
            .len = output.len,
            .data = output.ptr,
        };
        qtc.Attica__BuildServiceJobOutput_SetOutput(@ptrCast(self), output_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#output)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput, allocator: std.mem.Allocator ```
    pub fn Output(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__BuildServiceJobOutput_Output(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::buildservicejoboutput.Output: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#isRunning)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput ```
    pub fn IsRunning(self: ?*anyopaque) bool {
        return qtc.Attica__BuildServiceJobOutput_IsRunning(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#isCompleted)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput ```
    pub fn IsCompleted(self: ?*anyopaque) bool {
        return qtc.Attica__BuildServiceJobOutput_IsCompleted(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#isFailed)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput ```
    pub fn IsFailed(self: ?*anyopaque) bool {
        return qtc.Attica__BuildServiceJobOutput_IsFailed(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-buildservicejoboutput.html#isValid)
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.Attica__BuildServiceJobOutput_IsValid(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.Attica__BuildServiceJobOutput ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.Attica__BuildServiceJobOutput_Delete(@ptrCast(self));
    }
};

File: src_original/extras-attica/libbuildservicejoboutput.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_Attica__BuildServiceJobOutput
#include <QString>
#include <QByteArray>
#include <cstring>
#include <buildservicejoboutput.h>
#include "libbuildservicejoboutput.h"
#include "libbuildservicejoboutput.hxx"

Attica__BuildServiceJobOutput* Attica__BuildServiceJobOutput_new() {
    return new Attica::BuildServiceJobOutput();
}

Attica__BuildServiceJobOutput* Attica__BuildServiceJobOutput_new2(const Attica__BuildServiceJobOutput* other) {
    return new Attica::BuildServiceJobOutput(*other);
}

void Attica__BuildServiceJobOutput_OperatorAssign(Attica__BuildServiceJobOutput* self, const Attica__BuildServiceJobOutput* other) {
    self->operator=(*other);
}

void Attica__BuildServiceJobOutput_SetOutput(Attica__BuildServiceJobOutput* self, const libqt_string output) {
    QString output_QString = QString::fromUtf8(output.data, output.len);
    self->setOutput(output_QString);
}

libqt_string Attica__BuildServiceJobOutput_Output(const Attica__BuildServiceJobOutput* self) {
    QString _ret = self->output();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

bool Attica__BuildServiceJobOutput_IsRunning(const Attica__BuildServiceJobOutput* self) {
    return self->isRunning();
}

bool Attica__BuildServiceJobOutput_IsCompleted(const Attica__BuildServiceJobOutput* self) {
    return self->isCompleted();
}

bool Attica__BuildServiceJobOutput_IsFailed(const Attica__BuildServiceJobOutput* self) {
    return self->isFailed();
}

bool Attica__BuildServiceJobOutput_IsValid(const Attica__BuildServiceJobOutput* self) {
    return self->isValid();
}

void Attica__BuildServiceJobOutput_Delete(Attica__BuildServiceJobOutput* self) {
    delete self;
}
